forc-plugins/forc-doc/src/doc/mod.rs

```diff
@@ -76,16 +76,17 @@ impl Documentation {
         // currently this compares the spans as str, but this needs to change
         // to compare the actual types
         for doc in docs.iter_mut() {
-            let mut impl_vec: Vec<DocImplTrait> = Vec::new();
+            let mut impl_trait_vec: Vec<DocImplTrait> = Vec::new();
+            let mut inherent_impl_vec: Vec<DocImplTrait> = Vec::new();
```

Comment:
Just to make sure I got this right, is "inherent impl" impl for the struct itself? Like:

```sway
struct Foo {
 a: u64
}

// this is a inherent impl
impl Foo {

}
``` 

If so why this is a vec? is it because of generics?
```sway
script; 
struct Foo<T> {
 a: T
}

// this is a inherent impl
impl Foo<u64> {
 fn hey(a: u64) -> Self {
     Self {
         a
     }
 }
}

// this is an another inherent impl
impl Foo<u32> {
  fn hey2(a: u32) -> Self {
     Self {
         a
     }
 }
}
```

Reply:
Great question! That's exactly right. At first I thought about merging the inherent impl blocks but because of generics, they shouldn't be merged.